Modules/_sqlite/connection.c

```diff
@@ -958,6 +958,11 @@ step_callback(sqlite3_context *context, int argc, sqlite3_value **params)
     assert(ctx != NULL);
 
     aggregate_instance = (PyObject**)sqlite3_aggregate_context(context, sizeof(PyObject*));
+    if (aggregate_instance == NULL) {
+        (void)PyErr_NoMemory();
+        set_sqlite_error(context, "unable to allocate SQLite aggregate context");
+        goto error;
+    }
```

Comment:
If we had a way to hook into the SQLite memory allocator, we could simulate out-of-memory, but unfortunately the `sqlite3` extension module does not support this (yet).